packages/server/modules/auth/tests/apps.spec.js

```diff
@@ -214,6 +216,61 @@ describe('Services @apps-services', () => {
       .catch((err) => expect(err.message).to.equal('Access code not found.'))
   })
 
+  const defaultApps = ['spklwebapp', 'explorer', 'sdm', 'sca', 'spklexcel']
+  defaultApps.forEach((speckleAppId) => {
+    it(`Should not invalidate tokens, refresh tokens and access codes for default app: ${speckleAppId}, if updated`, async () => {
+      const { updateDefaultApp } = require('@/modules/auth/defaultApps')
```

Comment:
Better to keep all require calls near the top for clarity

Reply:
this one is a deliberate placement, the updateDefaultApp function used to be private. Since its operation on the default server apps it works with sensitive stuff, I would preferred to keep it private.

Do you have better options to testing not exported functions?

Reply:
The placement doesn't fully resolve the encapsulation issue, because devs can still import the newly public function anywhere. I get what you meant, you put the `require` call in a more hidden location so that devs don't see it, but if they just open up `defaultApps` directly they'll find the export there.

In this case you aren't testing the private function (which would be a code smell as well, since module/class internals should just be implementation details that you shouldn't need to directly test/validate), you're using a previously private function to prepare the test data state. Considering that you need it outside of the private module context, it can no longer be a private/non-exported function and it makes sense that you've exported it. It doesn't matter whether the new requirement comes from a test file or some business-oriented service - if the function needs to be public, it becomes public.

So the correct thing to do is still to move the `require` call to the top of the test file, the `updateDefaultApp` is a new public member of the `defaultApps` module.

I do get your concern regarding the semantics however - we want `updateDefaultApp` to be an internal function, not something to be used directly anywhere. For that problem proper organization of code and further separation of concerns is a good solution. You'll always have seemingly internal functions that have to be public/exported, just so you can invoke them from other seemingly internal areas of the app. And since that is the case, the only thing you can really do is organize internal things separately from public things. So, for example, instead of having default app definitions, service functions and module init code all defined in the same file (defaultApps.js), the service functions could be split out to a `defaultAppService` as exported members, and it should be somewhat clear to devs that this deals with the setup of default apps and is an "internal service".